### PR TITLE
Add tests for home planet meters on first turn

### PR DESCRIPTION
**Pull request by o01eg**
Tried to catch https://github.com/freeorion/freeorion/issues/3326 but actually it works on `master`.

### CODE REVIEW

File: test/system/SmokeTestGame.cpp [hunk 1]
```diff
@@ -153,6 +155,18 @@ BOOST_AUTO_TEST_CASE(host_server) {
...
+        if (m_current_turn == 1) {
+            // check home planet meters
+            bool found_planet = false;
+            for (const auto& planet : Objects().find<Planet>(OwnedVisitor(m_empire_id))) {
+                BOOST_REQUIRE_LT(0.0, planet->GetMeter(MeterType::METER_POPULATION)->Current());
+                BOOST_REQUIRE_LT(0.0, planet->GetMeter(MeterType::METER_INDUSTRY)->Current());
+                BOOST_REQUIRE_LT(0.0, planet->GetMeter(MeterType::METER_RESEARCH)->Current());
+                found_planet = true;
+            }
+            BOOST_REQUIRE(found_planet);
```
Comment (geoffthemedio) [inline test/system/SmokeTestGame.cpp:172]: Can these tests output something to the log when then pass?

**Comment by agrrr3** [comment]
@o01eg how do i run those tests on build? you are on ubuntu 18.04, right? it only seems to happen in my snapcraft environment

**Comment by o01eg** [comment]
@agrrr3  Not sure how to launch it with snapcraft. I could copy test to hostless system test so you can launch freeoriond with snap and make test use external server with `FO_TEST_HOSTLESS_LAUNCH_SERVER=0`

File: test/system/SmokeTestGame.cpp (hunk 1 above)
Comment (o01eg) [reply test/system/SmokeTestGame.cpp:172]: Yes, but it require `--log_level=message` command line argument

**Comment by agrrr3** [comment]
just a note. testing/debugging worked by adding -DBUILD_TESTING (cmake-parameters on core20, configflags on core18) and running snapcraft with --debug in order to drop into the snapcraft environment and calling fo_systemtest_game from there.

**Comment by geoffthemedio** [comment]
@o01eg could you rebase on latest master?

**Comment by o01eg** [comment]
Rebased.

**Comment by geoffthemedio** [comment]
The full client Travis build failed, possibly just timed out?

The headless tests fail.

Are these concerns?

**Comment by o01eg** [comment]
> The full client Travis build failed, possibly just timed out?

I didn't changed much, maybe cache cleaned by other builds.

> The headless tests fail.

```
unable to close socket due to exception: close: Bad file descriptor
```

That error I sometimes catch on Travis CI

> Are these concerns?

It doesn't look so.

**Pull request merged**